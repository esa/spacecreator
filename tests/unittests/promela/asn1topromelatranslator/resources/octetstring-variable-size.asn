MYMODULE DEFINITIONS ::= BEGIN

MyOctetString ::= OCTET STRING (SIZE (9 .. 13))

END