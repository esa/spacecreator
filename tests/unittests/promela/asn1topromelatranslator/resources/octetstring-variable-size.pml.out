inline OctetStringElement_generate_value(OctetStringElement_gv)
{
    int OctetStringElement_tmp;
    if
    ::  atomic {
        select (OctetStringElement_tmp : 0 .. 255);
    }
    fi;
    OctetStringElement_gv = OctetStringElement_tmp;
}
inline MyOctetString_length_generate_value(MyOctetString_length_gv)
{
    int MyOctetString_length_tmp;
    if
    ::  atomic {
        select (MyOctetString_length_tmp : 9 .. 13);
    }
    fi;
    MyOctetString_length_gv = MyOctetString_length_tmp;
}
inline MyOctetString_generate_value(MyOctetString_gv)
{
    int MyOctetString_it;
    MyOctetString_length_generate_value(MyOctetString_gv.length);
    for(MyOctetString_it : 0 .. (MyOctetString_gv.length - 1))
    {
        OctetStringElement_generate_value(MyOctetString_gv.data[MyOctetString_it]);
    }
}
